### ISSUE TITLE
Display XLF context information in Web interface

### ISSUE THREAD

**Issue by danjor** [WeblateOrg/weblate#7767]
### Describe the problem

Hello,

Given the trans-unit from XLF below :

```xml
<trans-unit id="9047795109808746529" datatype="html">
	<source>Hello</source>
	<target state="new">Hello</target>
	<context-group purpose="location">
	  <context context-type="sourcefile">src/a.html</context>
	  <context context-type="linenumber">251,253</context>
	</context-group>
	<context-group purpose="location">
	  <context context-type="sourcefile">src/b.html</context>
	  <context context-type="linenumber">742,744</context>
	</context-group>
</trans-unit>
```

I would expect to have contexts displayed in weblate web interface to be able to quickly find from where the translation come from in the code.

Is there a way to display this ?
Othewise it would be a useful feature.

### Describe the solution you'd like

To have sourcefile and linenumber would be helpful for translation.

**Comment by KaVeKa** [comment]
We're having the same inconvenience on our projects. :-) Would be really helpful to have this information displayed on the web interface.

**Comment by nijel** [comment]
First, translate-toolkit needs to support querying context groups by purpose here: https://github.com/translate/translate/blob/cc9a7b8fa4089399150a4605c1438c29897538bc/translate/storage/xliff.py#L542-L562. Once this is done, support in Weblate can be added.

**Comment by MartinTopfstedtSinc** [comment]
Hi,

i implemented the function for this into the translate-toolkit (https://github.com/translate/translate/pull/4948). 

i tried to include the new functionality into weblate, but i couldn't get the weblate development setup to run, so i didn't get a chance to test it directly in weblate, but here would be the code for weblate:

https://github.com/WeblateOrg/weblate/blob/6000024dcfa9a5c48248d17856f5ae33bcef1772/weblate/formats/ttkit.py#L640
`
@cached_property
    def locations(self):
        """Return comma separated list of locations."""
        locations = []
        for contextGroup in self.mainunit.getcontextgroupsbyattribute("purpose", "location"):
            for context in contextGroup:
                if len(context) == 2:
                    locations.append(context[0][1] + ":" + context[1][1])

        return ", ".join(x for x in locations if x is not None)
`

**Comment by nijel** [comment]
Will be fixed by https://github.com/WeblateOrg/weblate/pull/9583 once translate-toolkit with https://github.com/translate/translate/pull/4948 is released (the relase is blocked by https://github.com/translate/translate/pull/4966).

**Comment by louis-lau** [comment]
@nijel Is it supposed to display it like every letter is a separate location or am I missing some kind of configuration?
![image](https://github.com/WeblateOrg/weblate/assets/1346804/a7f849e4-3d51-4d98-b758-8da824548eaa)

The context-group looks like this straight out of angular:
```xml
<context-group purpose="location">
  <context context-type="sourcefile">src/app/app.component.html</context>
  <context context-type="linenumber">91</context>
</context-group>
```

**Comment by nijel** [comment]
This was fixed in translate-toolkit 3.10.1
